src/Microsoft.ML.Tokenizers/PreTokenizer/PreTokenizer.cs

```diff
@@ -15,14 +13,22 @@ namespace Microsoft.ML.Tokenizers
     /// in the original string. These offsets are in the `original` referential.
     /// It also contains any `Token` associated to the current split.
     /// </summary>
-    public readonly struct Split : IEquatable<Split>
+    public struct Split : IEquatable<Split>
     {
+        private readonly string? _originalString;
+        private string? _tokenString;
+
         /// <summary>
         /// Gets the underlying split token. Each SubString is represented by a token
         /// and in the end we might be carrying a lot of SubString representing various parts of the
         /// original input string.
         /// </summary>
-        public string TokenString { get; }
+        public string TokenString => _tokenString ??= _originalString!.Substring(Offset.Index, Offset.End - Offset.Index);
+
+        /// <summary>
+        /// Gets the underlying split token as a span.
+        /// </summary>
+        public ReadOnlySpan<char> TokenSpan => _tokenString is string s ? s.AsSpan() : _originalString.AsSpan(Offset.Index, Offset.End - Offset.Index);
```

Comment:
This is a public API addition. It's not fully realized yet, but once Model's APIs are able to accept spans, we can pass in TokenSpan rather than TokenString, and we'll save allocating a string per token.

Reply:
good idea!